zephyr/lib/regions_mm.c

```diff
@@ -507,6 +509,20 @@ void *vmh_alloc(struct vmh_heap *heap, uint32_t alloc_size)
 	return ptr;
 }
 
+/**
+ *  @brief Alloc function, reentrant
+ *	   see _vmh_alloc comment for details
+ */
+void *vmh_alloc(struct vmh_heap *heap, uint32_t alloc_size)
+{
+	k_spinlock_key_t key = k_spin_lock(&heap->lock);
```

Comment:
the commit message says, that "spinlocks are used as race here is not very likely," but the difference between mutexes and spinlocks is not only that spinlocks are "busy spinning" while waiting to acquire the lock, while mutexes go to sleep, allowing other work to be done while waiting, but also that spinlocks under Zephyr disable interrupts. Do we really want to disable interrupts while allocating from VMH?

Reply:
As I checked - Mutex is used in Zephyr's implementation of malloc. Changing to mutex